sway/config.c

```diff
@@ -549,43 +549,37 @@ static bool load_include_config(const char *path, const char *parent_dir,
 	return true;
 }
 
-bool load_include_configs(const char *path, struct sway_config *config,
+void load_include_configs(const char *path, struct sway_config *config,
 		struct swaynag_instance *swaynag) {
 	char *wd = getcwd(NULL, 0);
 	char *parent_path = strdup(config->current_config_path);
 	const char *parent_dir = dirname(parent_path);
 
 	if (chdir(parent_dir) < 0) {
-		free(parent_path);
-		free(wd);
-		return false;
+		sway_log(SWAY_ERROR, "failed to change working directory");
+		goto free_all;
 	}
 
 	wordexp_t p;
 
 	if (wordexp(path, &p, 0) != 0) {
-		free(parent_path);
-		free(wd);
-		return false;
+		goto out_wordexp;
 	}
 
 	char **w = p.we_wordv;
 	size_t i;
 	for (i = 0; i < p.we_wordc; ++i) {
 		load_include_config(w[i], parent_dir, config, swaynag);
 	}
-	free(parent_path);
 	wordfree(&p);
-
-	// restore wd
+out_wordexp:
+	// Attempt to restore working directory before returning.
 	if (chdir(wd) < 0) {
-		free(wd);
-		sway_log(SWAY_ERROR, "failed to restore working directory");
-		return false;
+		sway_log(SWAY_ERROR, "failed to change working directory");
 	}
-
+free_all:
```

Comment:
nitpick: can you change this to `cleanup` for consistency with the rest of the code base
(sorry missed this one before)